lib/semantics/check-select-stmt.cpp

```diff
@@ -0,0 +1,316 @@
+//===-- lib/semantics/check-select-stmt.cc --------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "check-select-stmt.h"
+#include "flang/parser/message.h"
+#include "flang/semantics/tools.h"
+
+namespace Fortran::semantics {
+
+class SelectCaseHelper {
+
+public:
+  SelectCaseHelper(parser::Messages &messages) : messages_{messages} {}
+
+  void ReportOverlapCaseValError(
+      parser::CharBlock &source1, parser::CharBlock &source2) {
+    // C1149
+    messages_
+        .Say(source2,
+            "SELECT CASE statement value at this location overlaps with the below location"_err_en_US)
+        .Attach(source1,
+            "SELECT CASE statement value overlaps with the above location"_err_en_US);
+  }
+
+  void Insert(std::int64_t intVal, parser::CharBlock source) {
+    auto selectCaseIntRangeValsIter =
```

Comment:
We conventionally use modern `{`braced`}` initialization for variables for safety, rather than the older `=` style.

Reply:
I have used the braces for initialization.